tabpfn/utils.py

```diff
@@ -201,16 +211,16 @@ def normalize_data(data, normalize_positions=-1):
 def remove_outliers(X, n_sigma=4, normalize_positions=-1):
     # Expects T, B, H
     assert len(X.shape) == 3, "X must be T,B,H"
-    #for b in range(X.shape[1]):
-        #for col in range(X.shape[2]):
+
     data = X if normalize_positions == -1 else X[:normalize_positions]
-    data_clean = data[:].clone()
+
     data_mean, data_std = torch_nanmean(data, axis=0), torch_nanstd(data, axis=0)
     cut_off = data_std * n_sigma
     lower, upper = data_mean - cut_off, data_mean + cut_off
 
-    data_clean[torch.logical_or(data_clean > upper, data_clean < lower)] = np.nan
-    data_mean, data_std = torch_nanmean(data_clean, axis=0), torch_nanstd(data_clean, axis=0)
+    mask = (data <= upper) & (data >= lower) & ~torch.isnan(data)
+    data_mean, data_std = torch_masked_mean_and_std(data, mask, axis=0)
```

Comment:
did you do a test that this does not change behavior? as this seems like a little bit of a dangerous change.

Reply:
I have written a test (test_standardization) for the function remove_outliers (and indirectly the new nanmean and nanstd) that I can push with the changes and remove afterward. What I find interesting is that torch.sum and torch.nansum return slightly different values, which are sometimes off after around the sixth digit of the number. In the test, I currently have rewritten the nansum to the normal sum, so that the results are equal.